### DISCUSSION TITLE
PDF files not opening in repositories

### DISCUSSION THREAD

**Discussion by iamsnehadas** [community/community#64419] [Repositories] [answered]
### Select Topic Area

Question

### Body

I have a repository in GitHub, where I had previously uploaded some pdf files. But, when I am trying to access any of the pdf files, it is showing that "Unable to render code block" and the download option is also not working. The same problem is happening if I am trying to access repositories of other users where pdf files are uploaded.
![Screenshot (1296)](https://github.com/community/community/assets/95439390/15f37383-c7f5-4078-a55e-e381f3896675)
![Screenshot (1295)](https://github.com/community/community/assets/95439390/d821f47b-ed9a-4077-87e5-c910c5f1dfaa)

I tried to open it both in GitHub website, and in my phone. But, neither of them is working. It is showing "Something went wrong" in GitHub mobile and "Unable to render code block" in the website. When I asked others to open the repository for me, it is working fine for them. What is the problem here and what is the solution? My repository link: https://github.com/iamsnehadas/Allpdfs

**Comment by 0xScratch** [comment]
For this issue, better try accessing the repository with a different browser or clearing your browser's cache, sometimes the issue is due to the browser
And Must report this issue to Github Support Team. Because Github uses a rendering engine to display certain file types and sometimes the rendering engine itself is encountered with some issues which leads to error 'Unable to render code block'. Github support team will be helpful in that!!

You can mark this as answer and close this discussion, and let me know if this issue got resolved by github or not..

**Comment by arafajs** [comment]
There are a few possible reasons why you are getting the "Unable to render code block" error when trying to access PDF files on GitHub.

- The PDF file may be too large. GitHub has a file size limit of 100MB for rendered files. If your PDF file is larger than this, it will not be able to be rendered.
- The PDF file may be in an unsupported format. GitHub only supports PDF files in the following formats: PDF, PostScript, and XPS. If your PDF file is in a different format, it will not be able to be rendered.
- There may be a temporary issue with GitHub. If you are seeing this error for all PDF files, regardless of their size or format, it is possible that there is a temporary issue with GitHub. Try refreshing the page or trying again later.

If you have ruled out the above possibilities, then the next thing you can try is to embed the PDF file in a GitHub page. This will allow you to view the PDF file without having to download it. To do this, follow these steps:

1. Go to the repository where the PDF file is located.
2. Click on the "Settings" tab.
3. Click on the "Pages" tab.
4. Click on the "Add Page" button.
5. In the "Source" field, select "File" and then enter the path to the PDF file.
6. Click on the "Create Page" button.
7. Once the page has been created, you can view the PDF file by clicking on the link in the page's contents.

If you are still having problems rendering PDF files on GitHub, you can contact GitHub support for help.

**Reply by iamsnehadas** [reply]
I cannot find the add page button here...
![Screenshot (1297)](https://github.com/community/community/assets/95439390/58a7d66b-7a1c-4845-9232-11cde07a329b)

**Reply by AhmedHamdadaELhadad** [reply]
not Fix any Things

**Comment by aymanxdev** [comment]
I've tested this across different browsers and the issue only arises in Safari. Specifically, I'm using Safari v14.

`SyntaxError: Unexpected private name #a. Cannot parse class method with private name.` [see this in the console]

The `Unexpected private name` error seems to be associated with JavaScript's private class fields, a feature that some older versions of Safari might not support. What browsers have you tried and version? https://caniuse.com/?search=private%20field

**Reply by iamsnehadas** [reply]
Chrome v116, Brave, Firefox, Bing ---everywhere it is giving the same problem.

**Comment by iamsnehadas** [comment]
I contacted GitHub support, they also could not identify the problem. It probably varies from device to device.

**Comment by henryonyeama** [comment]
I am currently having this same issue

**Comment by JeevanIITH** [comment]
Same I am too facing this issue from 2 months , I used edge and chrome browser . It is even not opening in GitHub mobile application .

**Comment by iamsnehadas** [comment]
Hey, so I tried different methods, but the problem was not entirely resolved. One possible solution can be using high-speed internet. When I was using my mobile internet hotspot, or was trying via the mobile application, it was not opening. But when I was using hi-speed wi-fi network, the files were opening. So it may be a network issue.

**Reply by iamsnehadas** [reply]
Also, the files are opening in other people's devices. So, I do not think there's any problem with the repository.

**Reply by LilanOvO** [reply]
yeah, I solve the issue by chaging my vpn.

**Comment by mjbernier** [comment]
Having the same issue; I get an error message "Error rendering embedded code ... Invalid PDF". This is a file that has been sitting out there for a while, and the only "embedded code" is some mailto: links for e-mail (it's a resume). The file opens just fine on my local PC. I tried re-rendering the file and uploading it again, but the error persists.

**Reply by henryonyeama** [reply]
What worked for me was uploading the ipynb file without having to convert it to pdf

**Comment by tsmcgrath** [comment]
I am having a similar issue but it's a 404. This file has been in the repository for some months and opens fine.
https://tsmcgrath.github.io/catch23/PDF/LowerComancheExample.pdf

But this one is a recent upload and I get a 404. I copy/pasted the path & name so there's no typo. Same location as the one that opens. Shows that it's there in the repo. Que pasa?
https://tsmcgrath.github.io/catch23/PDF/CatchmentBMP.pdf

**Reply by tsmcgrath** [reply]
I am getting this Github pages build error. But have no idea what the actual error is:

`Error:  Logging at level: debug GitHub Pages: github-pages v228 GitHub Pages: jekyll v3.9.3 Theme: jekyll-theme-primer Theme source: /usr/local/bundle/gems/jekyll-theme-primer-0.6.0 Requiring: jekyll-github-metadata Requiring: jekyll-seo-tag Requiring: jekyll-coffeescript Requiring: jekyll-commonmark-ghpages Requiring: jekyll-gist Requiring: jekyll-github-metadata Requiring: jekyll-paginate Requiring: jekyll-relative-links Requiring: jekyll-optional-front-matter Requiring: jekyll-readme-index Requiring: jekyll-default-layout Requiring: jekyll-titles-from-headings GitHub Metadata: Initializing... Source: /github/workspace/. Destination: /github/workspace/./_site Incremental build: disabled. Enable with --incremental Generating... EntryFilter: excluded /node_modules Generating: JekyllOptionalFrontMatter::Generator finished in 0.002682812 seconds. Generating: JekyllReadmeIndex::Generator finished in 0.005794525 seconds. Generating: Jekyll::Paginate::Pagination finished in 5.3e-06 seconds. Generating: JekyllRelativeLinks::Generator finished in 0.001869808 seconds. Generating: JekyllDefaultLayout::Generator finished in 0.000660303 seconds. Requiring: kramdown-parser-gfm Generating: JekyllTitlesFromHeadings::Generator finished in 0.017430274 seconds. Rendering: assets/css/style.scss Pre-Render Hooks: assets/css/style.scss Rendering Markup: assets/css/style.scss Rendering: InstallingArcGISPythonLibraries.md Pre-Render Hooks: InstallingArcGISPythonLibraries.md Rendering Markup: InstallingArcGISPythonLibraries.md Rendering Layout: InstallingArcGISPythonLibraries.md Layout source: theme GitHub Metadata: Generating for tsmcgrath/catch23 GitHub Metadata: Calling @client.repository("tsmcgrath/catch23", {:accept=>"application/vnd.github.drax-preview+json"}) GitHub Metadata: Calling @client.pages("tsmcgrath/catch23", {}) Rendering: markdowntests/5769623_1.md Pre-Render Hooks: markdowntests/5769623_1.md Rendering Markup: markdowntests/5769623_1.md Rendering Layout: markdowntests/5769623_1.md Layout source: theme Rendering: markdowntests/5769623_2.md Pre-Render Hooks: markdowntests/5769623_2.md Rendering Markup: markdowntests/5769623_2.md Rendering Layout: markdowntests/5769623_2.md Layout source: theme Rendering: markdowntests/Introduction.md Pre-Render Hooks: markdowntests/Introduction.md Rendering Markup: markdowntests/Introduction.md Rendering Layout: markdowntests/Introduction.md Layout source: theme Rendering: markdowntests/R081BY320TX.md Pre-Render Hooks: markdowntests/R081BY320TX.md Rendering Markup: markdowntests/R081BY320TX.md Rendering Layout: markdowntests/R081BY320TX.md Layout source: theme Rendering: markdowntests/R081BY323TX.md Pre-Render Hooks: markdowntests/R081BY323TX.md Rendering Markup: markdowntests/R081BY323TX.md Rendering Layout: markdowntests/R081BY323TX.md Layout source: theme Rendering: markdowntests/R081BY326TX.md Pre-Render Hooks: markdowntests/R081BY326TX.md Rendering Markup: markdowntests/R081BY326TX.md Rendering Layout: markdowntests/R081BY326TX.md Layout source: theme Rendering: markdowntests/R081BY334TX.md Pre-Render Hooks: markdowntests/R081BY334TX.md Rendering Markup: markdowntests/R081BY334TX.md Rendering Layout: markdowntests/R081BY334TX.md Layout source: theme Rendering: markdowntests/R081BY337TX.md Pre-Render Hooks: markdowntests/R081BY337TX.md Rendering Markup: markdowntests/R081BY337TX.md Rendering Layout: markdowntests/R081BY337TX.md Layout source: theme Rendering: markdowntests/R081BY343TX.md Pre-Render Hooks: markdowntests/R081BY343TX.md Rendering Markup: markdowntests/R081BY343TX.md Rendering Layout: markdowntests/R081BY343TX.md Layout source: theme Rendering: markdowntests/R081BY348TX.md Pre-Render Hooks: markdowntests/R081BY348TX.md Rendering Markup: markdowntests/R081BY348TX.md Rendering Layout: markdowntests/R081BY348TX.md Layout source: theme Rendering: markdowntests/R081BY350TX.md Pre-Render Hooks: markdowntests/R081BY350TX.md Rendering Markup: markdowntests/R081BY350TX.md Rendering Layout: markdowntests/R081BY350TX.md Layout source: theme Rendering: markdowntests/R081BY353TX.md Pre-Render Hook`

**Comment by 2intor** [comment]
This is very iritating issue. Still its not fixed. There are lot of usefults pdfs out there in git repos we are not able to access them. Shame!

**Comment by casino925** [comment]
my issue have soleved. 
You should change the option of settings chrome : privacy and security -web settings-contents- choice  "website can convey window and redirection".
sorry my chorme's languge is chinese, the settings may not correctly. but it does have this option can soleved the porblem.

**Reply by AhmedHamdadaELhadad** [reply]
not fix error

**Comment by ayanika02** [comment]
I am not sure if this is the reason, but I think I found a fix (worked for me at least). I renamed my pdf file so that it doesn't contain any space, and now I am not getting this error. Earlier, when my file name had a space, I kept getting this error. Renaming my file before uploading it to my repository worked.

**Comment by codemigs** [comment, marked as answer]
Turn of all extensions, then refresh, then add the extensions again one by one and see which extension might be causing this. For me, it was the my dark mode extension

**Comment by NeariX67** [comment]
The Extension "Truffle" on Firefox caused this for me.

**Reply by ezappala** [reply]
I've been wondering for the longest time why this was happening to me on every computer but only on firefox. This was the answer!

**Comment by mb720** [comment]
In Firefox using NoScript, I needed to allow the domain

    viewscreen.githubusercontent.com

for PDFs to get rendered.

**Comment by bulcke** [comment]
For me the culprit was https://github.com/marklieberman/foxygestures
